warden/group/manager_sql.go

```diff
@@ -88,58 +94,56 @@ func (m *SQLManager) GetGroup(id string) (*Group, error) {
 	}, nil
 }
 
-func (m *SQLManager) DeleteGroup(id string) error {
-	if _, err := m.DB.Exec(m.DB.Rebind("DELETE FROM hydra_warden_group WHERE id=?"), id); err != nil {
-		return errors.WithStack(err)
+func (m *SQLManager) deleteGroup(id string) func(tx *sqlx.Tx) error {
+	return func(tx *sqlx.Tx) error {
+		if _, err := tx.Exec(m.DB.Rebind("DELETE FROM hydra_warden_group WHERE id=?"), id); err != nil {
+			return err
+		}
+
+		return nil
 	}
-	return nil
 }
 
-func (m *SQLManager) AddGroupMembers(group string, subjects []string) error {
-	tx, err := m.DB.Beginx()
-	if err != nil {
-		return errors.Wrap(err, "Could not begin transaction")
-	}
+func (m *SQLManager) DeleteGroup(id string) error {
+	return m.applyInTransaction(m.deleteGroup(id))
+}
 
-	for _, subject := range subjects {
-		if _, err := tx.Exec(m.DB.Rebind("INSERT INTO hydra_warden_group_member (group_id, member) VALUES (?, ?)"), group, subject); err != nil {
-			if err := tx.Rollback(); err != nil {
-				return errors.WithStack(err)
+func (m *SQLManager) addGroupMembers(group string, subjects []string) func(tx *sqlx.Tx) error {
+	return func(tx *sqlx.Tx) error {
+		for _, subject := range subjects {
+			if _, err := tx.Exec(m.DB.Rebind("INSERT INTO hydra_warden_group_member (group_id, member) VALUES (?, ?)"), group, subject); err != nil {
+				if err := tx.Rollback(); err != nil {
+					return err
+				}
+				return err
```

Comment:
Normal errors do not have stack traces in go, to add a stack trace do `return errors.WithStack(err)`